### ISSUE TITLE
Replace existing Type Instantiation 

### ISSUE THREAD

**Issue by khionu** [RogueException/Discord.Net#868]
[The existing reflection code](https://github.com/RogueException/Discord.Net/blob/dev/src/Discord.Net.Commands/Utilities/ReflectionUtils.cs) does the same work as [ASP.NET DI's ObjectFactory](https://github.com/aspnet/DependencyInjection/blob/dev/shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs), which is already being pulled by using Microsoft.Extentions.DependencyInjection. It might be prudent to use the latter.

**Comment by quinchs** [comment]
@Cenngo Could you look at this?

**Comment by Cenngo** [comment]
> @Cenngo Could you look at this?

Yes, although ActivatorUtils.CreateIstance is 2 times slower than .Invoke it seems like activatorUtils.CreateFactory actually returns a compiled lambda expression which is much faster than .Invoke so its a good idea to implement this. Ill make a pr for this when i get the chance.

**Comment by Cenngo** [comment]
Actually i believe the ActivatorUtilities implementation doesnt support property injection by default, to avoid introducing a breaking change i might use the activation method from the Interaction Service
